### DISCUSSION TITLE
Updating Dockerfiles

### DISCUSSION THREAD

**Discussion by adriang-90** [argoproj/argo-cd#9048] [Ideas]
# Hello everyone !

I was looking over the github to find some good examples of **Dockerfiles** and I got a few questions about the things I found in this repos Dockerfiles:
 
1. Why do you use **ADD** instead of **COPY** in Dockerfiles? **COPY** is preferred and considered as a good practice.
2. Why you go for 
``` 
COPY go.mod go.mod
COPY go.sum go.sum
```
instead of:
```
WORKDIR /some_path
COPY go.* ./
```
3. Can I make a PR with changes I found useful?

**Comment by ishitasequeira** [comment]
@adriang-90 I am not sure why it has been used the way it is. But feel free to raise a PR with the changes you feel might be helpful.
